galsim/chromatic.py

```diff
@@ -1359,3 +1770,76 @@ def withScaledFlux(self, flux_ratio):
             return ChromaticAutoCorrelation( self.obj * math.sqrt(flux_ratio), **self.kwargs )
         else:
             return ChromaticObject(self).withScaledFlux(flux_ratio)
+
+class ChromaticOpticalPSF(ChromaticObject):
+    """A subclass of ChromaticObject meant to represent chromatic optical PSFs.
+
+    Chromaticity plays two roles in optical PSFs. First, it determines the diffraction limit, via
+    the wavelength/diameter factor.  Second, aberrations such as defocus, coma, etc. are typically
+    defined in physical distances, but their impact on the PSF depends on their size in units of
+    wavelength.  Other aspects of the optical PSF do not require explicit specification of their
+    chromaticity, e.g., once the obscuration and struts are specified in units of the aperture
+    diameter, their chromatic dependence gets taken care of automatically.
+
+    When using interpolation to speed up image rendering (see ChromaticObject.setupInterpolation()
+    method for details), including ~10-15 samples across any given bandpass should be sufficient.
+    For moderate accuracy, 5 is often sufficient.  Likewise, for high accuracy, `oversample_fac`
+    should be in the range 1.5-2, whereas for moderate accuracy, 1 is often sufficient.  All of
+    these statements assume that aberrations are not very large (typically <~0.25 waves, which is
+    commonly satisfied by space telescopes); if they are larger than that, then more stringent
+    settings are required.
+
+    Also, note that in order to render an image of a real physical object, the ChromaticOpticalPSF
+    should be convolved with some object that has an SED.  Drawing without doing so corresponds to
+    using a flat SED.
+
+    @param   diam          Telescope diameter in meters.
+    @param   aberrations   An array of aberrations, in nanometers.  The size and format of this
+                           array is described in the OpticalPSF docstring.
```

Comment:
I think it's only simple if you already understand how Airy diffraction works.  I'm not sure everyone using GalSim will remember their physics classes that covered this.

I was looking into where we document how to calculate lam_over_diam and it's really only in Airy that there is a concrete annotated example.  (Also demo3 and demo9 do the calculation.)  So I think a similar explanation/example should be added to OpticalPSF and ChromaticOpticalPSF, since this is kind of a confusing thing.  

However, I've kind of grown fond of the idea of letting GalSim do the `lam * 1.e-9 / diam * 206265` calculation for the user given `lam` and `diam` in their natural units (nm and m).  So could we add that as an option in all these places, but keep the `lam_over_diam` option as well for people who want to use different units or something?  Then in ChromaticOpticalPSF, this would also serve as the fiducial wavelength for the `aberrations`.